### PR TITLE
Collabora for ns8

### PR DESCRIPTION
**Pull request by stephdl**
Make an User interface and a container to use collabora with NS8 and nextcloud

install nextcloud in ns8
then install richdocuments by the web interface or ssh to nextcloud and install it by hand

```
ssh nextcloud1@localhost
.config/bin/occ  app:install richdocuments
.config/bin/occ config:app:set richdocuments wopi_url --value=https://collabora.domain.com
.config/bin/occ app:enable richdocuments
```
Enventually we could make a checkbox and an inout box inside nextcloud to enable and set the url to collabora.

What do you think ?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,22 +19,34 @@ Output example:
...
+  "admin_password": "password"
+}
+EOF
+```
 
 The above command will:
 - start and configure the collabora instance
-- (describe configuration process)
-- ...
+- configure a virtual host for trafik to access the instance
```
Comment (gsanchietti) [inline README.md:42]:
```suggestion
- configure a virtual host for traefik to access the instance
```

File: imageroot/actions/configure-module/validate-input.json [hunk 2]
```diff
@@ -1,8 +1,42 @@
...
+            "description": "Request a valid Let's Encrypt certificate."
+        },
+        "http2https": {
+            "type": "boolean",
+            "title": "HTTP to HTTPS redirection",
+            "description": "Redirect all the HTTP requests to HTTPS"
+        },
+        "admin_password":{
+            "type": "string",
+            "description": "Default password for administrator user, it can be changed later"
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/validate-input.json:39]: Maybe it's better to set a minimum length of 1 char.

File: imageroot/actions/get-configuration/20read [hunk 3]
```diff
@@ -19,11 +19,22 @@
...
 import agent
 
+# Prepare return variable
 config = {}
 
+# Read current configuration from Redis
+env = f'module/{os.environ["MODULE_ID"]}/environment'
+rdb = agent.redis_connect()
+
+config["admin_password"] =  rdb.hget(env, "ADMIN_PASSWORD");
```
Comment (gsanchietti) [inline imageroot/actions/get-configuration/20read:34]: Just an idea: you get the value directly from the ENV without accessing redis at all

File: imageroot/systemd/user/collabora.service [hunk 4]
```diff
@@ -2,25 +2,26 @@
...
-    --cidfile=%t/collabora.ctr-id \
-    --cgroups=no-conmon \
-    --replace --name=%N \
-    --publish=127.0.0.1:${TCP_PORT}:8080 \
-    ${ECHO_SERVER_IMAGE}
-ExecStop=/usr/bin/podman stop --ignore --cidfile %t/collabora.ctr-id -t 10
-ExecStopPost=/usr/bin/podman rm --ignore -f --cidfile %t/collabora.ctr-id
+TimeoutStopSec=70
+ExecStartPre=/bin/rm -f %t/collabora.pid %t/collabora.pod-id
+ExecStartPre=/usr/bin/podman pod create --infra-conmon-pidfile %t/collabora.pid \
```
Comment (gsanchietti) [inline imageroot/systemd/user/collabora.service:17]: The app is composed only by one container: you really do not need a pod.

File: imageroot/actions/configure-module/validate-input.json (hunk 2 above)
Comment (stephdl) [reply imageroot/actions/configure-module/validate-input.json:40]: done

File: README.md (hunk 1 above)
Comment (stephdl) [reply README.md]: done

File: imageroot/actions/get-configuration/20read (hunk 3 above)
Comment (stephdl) [reply imageroot/actions/get-configuration/20read]: done

File: imageroot/actions/configure-module/01_validate_domain [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+# set error validation
+agent.set_weight(os.path.basename(__file__), 0)
+
+
+
+http = 0
+https = 0
+
+def testUrl(domain):
```
Comment (gsanchietti) [inline imageroot/actions/configure-module/01_validate_domain:49]: Why using a function for this?
You just call it once.
Also I see no reason to use global vars.

File: imageroot/actions/get-configuration/20read [hunk 6]
```diff
@@ -19,11 +19,19 @@
...
 import sys
 import agent
 
+# Prepare return variable
 config = {}
 
+config["admin_password"] =  os.environ.get("ADMIN_PASSWORD",'')
+config["host"] = os.environ.get("TRAEFIK_HOST",'')
+config["http2https"] =  os.environ.get("TRAEFIK_HTTP2HTTPS",False) == "True"
+config["lets_encrypt"] =  os.environ.get("TRAEFIK_LETS_ENCRYPT",False) == "True"
```
Comment (gsanchietti) [inline imageroot/actions/get-configuration/20read:33]: It's better to use always the same type.
```suggestion
config["http2https"] =  os.environ.get("TRAEFIK_HTTP2HTTPS","False") == "True"
config["lets_encrypt"] =  os.environ.get("TRAEFIK_LETS_ENCRYPT","False") == "True"
```

File: imageroot/actions/configure-module/01_validate_domain (hunk 5 above)
Comment (stephdl) [reply imageroot/actions/configure-module/01_validate_domain]: done

**Review by gsanchietti — APPROVED** [review]

**Comment by gsanchietti** [comment]
> Enventually we could make a checkbox and an inout box inside nextcloud to enable and set the url to collabora.

I really like the idea!

**Pull request merged**